### PR TITLE
Authentication prototype part 2

### PR DESCRIPTION
**Pull request by alex-eg**

### CODE REVIEW

File: backend/src/handlers/user.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct UserResponse {
+    pub id: Uuid,
+    pub handle: String,
+    pub email: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct UsersResponse(pub Vec<UserResponse>);
```
[the remark this answers is not in the document]
Comment (alex-eg) [reply backend/src/handlers/user.rs:17]: Nope, it's single response, containing a number of users.

File: backend/src/database.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        info!("User database found in {}, loading", path);
+        let lines = fs::read_to_string(path).expect(&format!("Failed to read {}", path));
+        serde_json::from_str(&lines).unwrap()
+    } else {
+        info!("User database not found, creating default in {}", path);
+        let number_of_users = 5;
+        let users = create_user_storage(number_of_users);
+        let string = serde_json::to_string(&users).unwrap();
+        let mut file = fs::File::create(path).unwrap();
+        file.write_all(&string.into_bytes()).unwrap();
```
[the remark this answers is not in the document]
Comment (alex-eg) [reply backend/src/database.rs:37]: We can, but we don't need to. If we can't write to file for whatever reason, we're already fucked.

File: backend/src/database.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+use std::{fs, path};
+
+// Stub user data storage
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UserStorage(pub Vec<User>);
+
+lazy_static! {
+    pub static ref USER_STORAGE: UserStorage = load_user_storage();
```
[the remark this answers is not in the document]
Comment (alex-eg) [reply backend/src/database.rs:12]: Just like our `CONFIG`: the storage will be initialized on the first access.

File: backend/src/errors.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+impl ResponseError for ApiError {
+    fn error_response(&self) -> HttpResponse {
+        match self {
+            _ => HttpResponse::new(StatusCode::INTERNAL_SERVER_ERROR),
+        }
+    }
+}
+
+impl From<BlockingError<ApiError>> for ApiError {
```
[the remark this answers is not in the document]
Comment (alex-eg) [reply backend/src/errors.rs:36]: Yes, in the `get_user` and `get_users` functiuons.

**Comment by alex-eg** [comment]
>I think, it will be better to put some files into the directories

For example?

>Tests are needed

Yes, they are, but I'd want to write for our new framework, and it hasn't merged yet.

File: backend/src/handlers/user.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+/// Api
+/// Get user
+pub async fn get_user(
+    data: Data<UserStorage>,
+    user_id: Path<Uuid>,
+) -> Result<Json<UserResponse>, ApiError> {
+    info!("User id: {}", *user_id);
+    let user = block(move || {
+        let user = find(&data, *user_id);
```
Comment (bronvic) [inline backend/src/handlers/user.rs:43]: maybe just _match find(...) {_

**Pull request merged**